src/services/shard-client.ts

```diff
@@ -0,0 +1,64 @@
+import { CombinedGrantsExtQuery } from 'context/PreFetchData';
+import { ShardClient, ShardSelector } from 'data-plane-gateway';
+import getGatewayAuthConfig from 'services/gateway-auth-config';
+import { SetShards } from 'stores/ShardDetail';
+import { LiveSpecsExtBaseQuery } from 'types';
+import {
+    removeGatewayAuthConfig,
+    storeGatewayAuthConfig,
+} from 'utils/env-utils';
+
+enum ErrorFlags {
+    TOKEN_NOT_FOUND = 'Unauthenticated',
+    TOKEN_INVALID = 'Authentication failed',
+    OPERATION_INVALID = 'Unauthorized',
+}
+
+const getShardList = <T extends LiveSpecsExtBaseQuery>(
+    baseUrl: URL,
+    authToken: string,
+    specs: T[],
+    setShards: SetShards,
+    sessionKey: string,
+    grantDetails: CombinedGrantsExtQuery[]
+) => {
+    const shardClient = new ShardClient(baseUrl, authToken);
+    const taskSelector = new ShardSelector();
+
+    specs
+        .map((spec) => spec.catalog_name)
+        .forEach((name) => taskSelector.task(name));
+
+    shardClient
+        .list(taskSelector)
+        .then((result) => {
+            const shards = result.unwrap();
+
+            if (shards.length > 0) {
+                setShards(shards);
+            }
+        })
+        .catch((error: string) => {
+            if (
+                error.includes(ErrorFlags.TOKEN_INVALID) ||
+                error.includes(ErrorFlags.TOKEN_NOT_FOUND) ||
+                error.includes(ErrorFlags.OPERATION_INVALID)
```

Comment:
This conditional statement needs to be narrowed, removing the check for invalid operation flag. The check was included as a temporary patch due to the deployment of a stale version of the `data-plane-gateway`.

Reply:
Anything bad happen by leaving it in?

Reply:
No, it will just result in the regeneration of a valid token. Since the `data-plane-gateway` was redeployed, I will evaluate whether it contains the changes needed to remove the extra condition.